pkg/cri/server/metrics.go

```diff
@@ -0,0 +1,62 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package server
+
+import "github.com/docker/go-metrics"
+
+const (
+	// unicorns and rainbows
+	networkOKStatus = "ok"
+	// no cni config
+	networkNotInitializedErrorStatus = "network_config_not_initialized_error"
+	// parsing annotations and other options failed
+	networkOptionsErrorStatus = "network_options_parse_error"
+	// cni op (e.g. ADD/DEL) failed
+	networkOpErrorStatus = "network_op_error"
+	// Used on network setup when create was successful but we couldn't find the network in the result
+	networkNotFoundErrorStatus = "network_not_found_error"
+	networkSandboxCreateError  = "sandbox_create_error"
+)
+
+type collector struct {
+	networkSetupLatency         metrics.LabeledTimer
+	networkTeardownLatency      metrics.LabeledTimer
+	networkSandboxCreateLatency metrics.LabeledTimer
+}
+
+func (c *criService) configureMetrics(ns *metrics.Namespace) {
+	c.metrics.networkSetupLatency = ns.NewLabeledTimer("network_setup", "The time it takes to configure the network sandbox for a pod", "status")
+	c.metrics.networkTeardownLatency = ns.NewLabeledTimer("network_teardown", "The time it takes to teardown a pod's network sandbox", "status")
+
+	for _, l := range []string{
```

Comment:
I'm wondering if these are too much

Reply:
it does seem like overkill given anyone looking at these would probably also be looking at error logs and / or current reported status..